docs/pages/deploy-a-cluster/high-availability.mdx

```diff
@@ -0,0 +1,369 @@
+    region: "us-east-1" # Or another region
+    table_name: "my-kv-table"
+    audit_events_uri: ["dynamodb://my-audit-events-table", "stdout://"]
+    continuous_backups: true
+    audit_sessions_uri: "s3://my-session-recording-bucket"
+  auth_service:
+    enabled: true
+    cluster_name:
+      "mycluster.example.com"
+      # Remove this if not using Teleport Enterprise
+    license_file: "/var/lib/license/license.pem"
+  proxy_service:
+    public_addr: "mycluster.example.com:443"
+    enabled: true
+    https_keypairs:
+      - key_file: /etc/teleport-tls/tls.key
+        cert_file: /etc/teleport-tls/tls.crt
+  ssh_service:
+    enabled: false
+```
+</TabItem>
+<TabItem label="Google cloud Firestore">
+```yaml
+version: v1
+teleport:
+  storage:
+    type: firestore
+    project_id: my-google-cloud-project
+    collection_name: my-cluster-state-table
+    credentials_path: /etc/teleport-secrets/gcp-credentials.json
+    audit_events_uri: ['firestore://my-audit-events-table?projectID=my-google-cloud-project&credentialsPath=/etc/teleport-secrets/gcp-credentials.json', 'stdout://']
+    audit_sessions_uri: "gs://session-recording-bucket?projectID=my-google-cloud-project&credentialsPath=/etc/teleport-secrets/gcp-credentials.json"
+  auth_service:
+    enabled: true
+    cluster_name: "mycluster.example.com"
+    # Remove this if not using Teleport Enterprise
+    license_file: "/var/lib/license/license.pem"
+  proxy_service:
+    public_addr: "mycluster.example.com:443"
+    enabled: true
+    https_keypairs:
+    - key_file: /etc/teleport-tls/tls.key
+      cert_file: /etc/teleport-tls/tls.crt
+  ssh_service:
+    enabled: false
+```
+</TabItem>
+<TabItem label="etcd">
+```yaml
+version: v1
+teleport:
+  storage:
+    type: etcd
+    peers: ["https://172.17.0.1:4001", "https://172.17.0.2:4001"]
+    tls_cert_file: /var/lib/teleport/etcd-cert.pem
+    tls_key_file: /var/lib/teleport/etcd-key.pem
+    tls_ca_file: /var/lib/teleport/etcd-ca.pem
+    username: username
+    password_file: /mnt/secrets/etcd-pass
+    prefix: /teleport/
+    insecure: false
+    etcd_max_client_msg_size_bytes: 15728640
+    audit_sessions_uri: "s3://my-session-recording-bucket"
+  auth_service:
+    enabled: true
+    cluster_name: "mycluster.example.com"
+    # Remove this if not using Teleport Enterprise
+    license_file: "/var/lib/license/license.pem"
+  proxy_service:
+    public_addr: "mycluster.example.com:443"
+    enabled: true
+    https_keypairs:
+    - key_file: /etc/teleport-tls/tls.key
+      cert_file: /etc/teleport-tls/tls.crt
+  ssh_service:
+    enabled: false
+```
+</TabItem>
+</Tabs>
+
+This configuration has the following required settings:
+
+- **Disable TLS multiplexing:** This configuration uses the `v1` version of the
+  Teleport configuration (`version: v1`) to disable [TLS
+  multiplexing](../management/operations/tls-routing.mdx), which is required
+  when running Teleport with a load balancer in high-availability mode. 
```

Comment:
I don't think this is required nor a good practice. v1 keeps multiplexing disabled for compatibility reasons, and the Helm charts defaults to separate listeners to stay backward compatible for users running on EKS with ACM and a LB terminating TLS on 443.

Reply:
IIUC using `version: v1` in the config does two things:

- disables TLS multiplexing, equivalent to doing this on `v2`+:
```yaml
auth_service:
  proxy_listener_mode: separate
```
- automatically enables the legacy listeners on separate ports for any enabled services:
```yaml
proxy_service:
  enabled: true
  listen_addr: 0.0.0.0:3023
  tunnel_listen_addr: 0.0.0.0:3024
```

Essentially it's a shortcut to doing less legwork in the config file. It's currently the easiest way I know of to get a "legacy"/separate port config without having to be very explicit about it.